src/main/java/com/synopsys/integration/detect/configuration/DetectProperties.java

```diff
@@ -936,6 +936,15 @@ private DetectProperties() {
             .setGroups(DetectGroup.IMPACT_ANALYSIS, DetectGroup.GLOBAL, DetectGroup.SOURCE_SCAN)
             .build();
 
+    public static final BooleanProperty DETECT_COMPONENT_LOCATION_ANALYSIS_ENABLED =
+            BooleanProperty.newBuilder("detect.component.location.analysis.enabled", false)
+                    .setInfo("Component Location Analysis Enabled", DetectPropertyFromVersion.VERSION_8_8_0) // TODO update Detect version before merging FIX PR feature branch to master
+                    .setHelp(
+                            "If set to true, Detect will generate an output file detailing where dependencies are declared.",
+                            "All dependencies will be included when using Synopsys Detect in offline mode. Only policy violating dependencies will be included for Rapid and Stateless Scan modes.")
+                    .setGroups(DetectGroup.GENERAL)
+                    .build();
```

Comment:
- Do we want to specify the name of the file/where this file is saved?
- Is using "components" and "dependencies" interchangeably a bad idea? If so, which should be our word of choice to minimize confusion/be uniform with rest of our documentation? See [here](https://community.synopsys.com/s/document-item?bundleId=integrations-detect&topicId=properties/detectors/conan.html&anchor=attempt-package-revision-match&_LANG=enus) for an example of where the two words are referring to slightly different things.

Reply:
Hi Shanty.

Yes please on specifying the filename and location. That will be very helpful for users.

It would be good to standardize on using as accurate of terms as we can, though it can get blurry as to where each should be employed, and there is a lot of overlap. (It gets even messier when you throw the term "library" into the mix.)
We may want to look at it this way; use the term "component" when not specifically referring to something we have identified as being required to interact with other components for their compilation or function, OR when generally listing findings that may or may not be required for provision of other components' functionality.
Then use the term "dependency" where we are indicating that this specified component is a requirement for the functionality of another identified component.
(Ya, I probably made that all sound more complex than it is.)

For location, it looks like it should also go under this configuration section as a configuration property? https://sig-product-docs.synopsys.com/bundle/integrations-detect/page/properties/configuration/overview.html

Reply:
1. 8.11?
2. Yes, with output path as anchor.
3. Use component in general. Use dependency to emphasize hierarchal relation when it matters.
4. +1 to what Chris said above.